Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -1166,18 +1122,65 @@ namespace OpenInfraPlatform {
 
 							double radiusMax = std::max(xRadius, yRadius);
 							double radiusMin = std::min(xRadius, yRadius);
-							int num_segments = GeomSettings()->getNumberOfSegmentsForTessellation(radiusMin);
+							int numSegments = GeomSettings()->getNumberOfSegmentsForTessellation(radiusMin);
 							double deltaAngle = GeomSettings()->getAngleLength(radiusMin);
 
+							carve::geom::vector<3> ellipse_center =
+								conicPositionMatrix * carve::geom::VECTOR(0, 0, 0);
+
 							// todo: implement clipping
-							if (!trim1Vec.empty() || !trim2Vec.empty())
-								// Internal TODO: Implement function GetPointOnCurve, which will be able calculate trimming for each curve
-								throw oip::InconsistentModellingException(ellipse, "Trimming not supported");
+							if (!trim1Vec.empty() || !trim2Vec.empty()) {
+								//Calculate an angle on the ellipse for trimming begin.
+								double startAngle = 0.;
+								for (const auto& element : trim1Vec)
+								{
+									carve::geom::vector<3> point = getPointOnCurve(ellipse, *element);
+									startAngle = getAngleOnEllipse(ellipse_center, xRadius, yRadius, point);
+								}
+								//Calculate an angle on the ellipse for trimming end.
+								double endAngle = 0.;
+								for (const auto& element : trim2Vec)
+								{
+									carve::geom::vector<3> point = getPointOnCurve(ellipse, *element);
+									endAngle = getAngleOnEllipse(ellipse_center, xRadius, yRadius, point);
+								}
+								// Calculate an opening angle
+								double openingAngle = 0.;
+								if (senseAgreement) {
+									if (startAngle < endAngle) {
+										openingAngle = endAngle - startAngle;
+									}
+									else {
+										// circle passes 0 angle
+										openingAngle = endAngle - startAngle + 2.0*M_PI;
+									}
+								}
+								else {
+									if (startAngle > endAngle) {
+										openingAngle = endAngle - startAngle;
+									}
+									else {
+										// circle passes 0 angle
+										openingAngle = endAngle - startAngle - 2.0*M_PI;
+									}
+								}

```

Comment:
This seems to be doubling of code from `IfcCircle` - please conform to the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). Put this in a function.

Reply:
Resolved: c6c7f78